### PR TITLE
fix: dockerfile ui environment variables

### PR DESCRIPTION
**Pull request by lorenzodagostinoradicalbit**
Actually ui images (nginx) are broken because is not possible to dynamically set VITE_BASE_URL variables.

This PR fix the issue by manually changing the base url in builded project using environment variables.

Also added a new stage in Dockerfile (dev) which is used in docker-compose commands.

*Notes*:
- environment variables must start with APP_ prefix.
- actually only production build are broken. Starting the app via docker compose is working

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "start:dev": "NODE_ENV=development && vite --host 0.0.0.0",
     "build:local": "NODE_ENV=local && vite build",
     "build:dev": "NODE_ENV=development && vite build",
+    "build:prod": "vite build --mode production",
```
Comment (paoloyx) [inline ui/package.json:11]: If we put `NODE_ENV=production && vite build` is the same? Just wondering

File: ui/Dockerfile [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
 FROM nginx:alpine3.19
 
 EXPOSE 80
 
-COPY --from=build /app/dist /usr/share/nginx/html
-COPY nginx.conf /etc/nginx/conf.d/default.conf
\ No newline at end of file
+COPY --from=build /app/dist /usr/share/nginx/html/
+COPY nginx.conf /etc/nginx/conf.d/default.conf
+COPY env.sh /docker-entrypoint.d/
```
Comment (paoloyx) [inline ui/Dockerfile:25]: so, every file that exists into the `/docker-entrypoint.d` is executed when the OCI image is started?

**Review by paoloyx — APPROVED** [review]

File: ui/Dockerfile (hunk 2 above)
Comment (lorenzodagostinoradicalbit) [reply ui/Dockerfile:25]: Correct 👍

**Comment by devops-radicalbit** [comment]
### BOT: Running Job
Requested JOB status is available [here](https://github.com/radicalbit/radicalbit-ai-monitoring/actions/runs/9709461566)

**Comment by devops-radicalbit** [comment]
### BOT: Running Job
Requested JOB status is available [here](https://github.com/radicalbit/radicalbit-ai-monitoring/actions/runs/9709501044)

**Pull request merged**